Add common-security-config modules to dependency management

We are currently missing the following modules in the Dataflow dependency management (`/spring-cloud-dataflow/spring-cloud-dataflow-dependencies/pom.xml`)

- spring-cloud-common-security-config-core
- spring-cloud-common-security-config-web
- spring-cloud-starter-common-security-config-web

Additionally, we are also using `project.version` as the dependency version in other modules in Dataflow, such as `/spring-cloud-dataflow/spring-cloud-dataflow-server-core/pom.xml`:
```xml
<dependency>
    <groupId>org.springframework.cloud</groupId>
    <artifactId>spring-cloud-common-security-config-web</artifactId>
    <version>${project.version}</version>
</dependency>
<dependency>
    <groupId>org.springframework.cloud</groupId>
    <artifactId>spring-cloud-starter-common-security-config-web</artifactId>
    <version>${project.version}</version>
    <type>test-jar</type>
    <scope>test</scope>
    <exclusions>
        <exclusion>
            <groupId>org.codehaus.jackson</groupId>
            <artifactId>jackson-mapper-asl</artifactId>
        </exclusion>
    </exclusions>
</dependency>
```
In normal conditions this is not a problem, the Dataflow project.version is used and all is well. However, if an application extends Dataflow and pulls the above module into its build, the project version of the consuming project will be used. 

Example: SCDF Pro has a reference to
```xml
<dependency>
    <groupId>com.vmware.dataflow</groupId>
    <artifactId>scdf-pro-server-core</artifactId>
</dependency>
```
This ends up pulling in the `common-security-config-web` and uses a version of `1.6.0-SNAPSHOT` (which is the Pro project.version) which just so happens that the `common-security-config-web` has an old version number of 1.6.x and so it uses the very old 1.6.x dependencies.

### Re-evaluate usage of project.version
We should also re-evaluate our usage of project.version as our version identifier. Where else could this be an issue for us?

More specifically this is an issue when another project uses SCDF parent module as their parent. 

Given the following simple app pom.xml:
```xml
<?xml version="1.0" encoding="UTF-8"?>
<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 https://maven.apache.org/xsd/maven-4.0.0.xsd">
    <modelVersion>4.0.0</modelVersion>
    <parent>
        <groupId>org.springframework.cloud</groupId>
        <artifactId>spring-cloud-dataflow-parent</artifactId>
        <version>2.11.0</version>
        <relativePath/>
    </parent>
    <groupId>com.example</groupId>
    <artifactId>demo1</artifactId>
    <version>0.0.1-SNAPSHOT</version>

    <properties>
        <java.version>17</java.version>
    </properties>

    <dependencyManagement>
        <dependencies>
            <dependency>
                <groupId>org.springframework.cloud</groupId>
                <artifactId>spring-cloud-dataflow-dependencies</artifactId>
                <version>2.11.0</version>
                <type>pom</type>
                <scope>import</scope>
            </dependency>
            <dependency>
                <groupId>org.springframework.boot</groupId>
                <artifactId>spring-boot-dependencies</artifactId>
                <version>2.7.16</version>
                <type>pom</type>
                <scope>import</scope>
            </dependency>
        </dependencies>
    </dependencyManagement>

    <dependencies>
        <dependency>
            <groupId>org.springframework.boot</groupId>
            <artifactId>spring-boot-starter</artifactId>
        </dependency>
        <dependency>
            <groupId>org.springframework.cloud</groupId>
            <artifactId>spring-cloud-starter-dataflow-server</artifactId>
        </dependency>
        <dependency>
            <groupId>org.springframework.boot</groupId>
            <artifactId>spring-boot-starter-test</artifactId>
            <scope>test</scope>
        </dependency>
    </dependencies>

    <build>
        <plugins>
            <plugin>
                <groupId>org.springframework.boot</groupId>
                <artifactId>spring-boot-maven-plugin</artifactId>
            </plugin>
        </plugins>
    </build>

</project>
```
Will fail w/  
```
[ERROR] The build could not read 1 project -> [Help 1]
[ERROR]   
[ERROR]   The project com.example:demo1:0.0.1-SNAPSHOT (/Users/cbono/Desktop/demo1/pom.xml) has 2 errors
[ERROR]     Non-resolvable import POM: Could not find artifact org.springframework.cloud:spring-cloud-dataflow-build-dependencies:pom:0.0.1-SNAPSHOT @ org.springframework.cloud:spring-cloud-dataflow-parent:2.11.0, /Users/cbono/.m2/repository/org/springframework/cloud/spring-cloud-dataflow-parent/2.11.0/spring-cloud-dataflow-parent-2.11.0.pom, line 104, column 16 -> [Help 2]
[ERROR]     Non-resolvable import POM: Could not find artifact org.springframework.cloud:spring-cloud-dataflow-common-dependencies:pom:0.0.1-SNAPSHOT @ org.springframework.cloud:spring-cloud-dataflow-parent:2.11.0, /Users/cbono/.m2/repository/org/springframework/cloud/spring-cloud-dataflow-parent/2.11.0/spring-cloud-dataflow-parent-2.11.0.pom, line 111, column 16 -> [Help 2]
```
as `org.springframework.cloud:spring-cloud-dataflow-build-dependencies:pom:0.0.1-SNAPSHOT` is not a valid artifact. The cause of this is that the parent uses these version variables that are set w/ `project.version`. 
```xml
<dataflow.version>${project.version}</dataflow.version>
<spring-cloud-dataflow-common.version>${project.version}</spring-cloud-dataflow-common.version>
<spring-cloud-skipper.version>${project.version}</spring-cloud-skipper.version>
```
If I define the following in my demo app above: 
```xml
<dataflow.version>2.11.0</dataflow.version>
<spring-cloud-dataflow-common.version>2.11.0</spring-cloud-dataflow-common.version>
<spring-cloud-skipper.version>2.11.0</spring-cloud-skipper.version>
```
then I get to the original symptom reported in this issue: 
```
[ERROR] Failed to execute goal on project demo1: Could not resolve dependencies for project com.example:demo1:jar:0.0.1-SNAPSHOT: Could not find artifact org.springframework.cloud:spring-cloud-common-security-config-web:jar:0.0.1-SNAPSHOT -> [Help 1]
```

--- a/.github/workflows/milestone-worker.yml
+++ b/.github/workflows/milestone-worker.yml
@@ -75,6 +75,13 @@ jobs:
           -gs .github/settings.xml \
           -Pstagingmilestone \
           -DgenerateBackupPoms=false \
+          -Dproperty=dataflow.version \
+          -DnewVersion=${BUILD_ZOO_HANDLER_spring_cloud_dataflow_version} \
+          -B        
+        jfrog rt mvn versions:set-property \
+          -gs .github/settings.xml \
+          -Pstagingmilestone \
+          -DgenerateBackupPoms=false \
           -Dproperty=spring-cloud-deployer.version \
           -DnewVersion=${BUILD_ZOO_HANDLER_spring_cloud_deployer_version} \
           -B

--- a/.github/workflows/next-dev-version-worker.yml
+++ b/.github/workflows/next-dev-version-worker.yml
@@ -69,6 +69,11 @@ jobs:
         echo BUILD_ZOO_HANDLER_spring_cloud_dataflow_version=$(mvn help:evaluate -Dexpression=project.version -q -DforceStdout) >> $GITHUB_ENV
         jfrog rt mvn versions:set-property \
           -DgenerateBackupPoms=false \
+          -Dproperty=dataflow.version \
+          -DnewVersion=${BUILD_ZOO_HANDLER_spring_cloud_dataflow_version} \
+          -B        
+        jfrog rt mvn versions:set-property \
+          -DgenerateBackupPoms=false \
           -Dproperty=spring-cloud-deployer.version \
           -DnewVersion=${BUILD_ZOO_HANDLER_spring_cloud_deployer_version} \
           -B

--- a/.github/workflows/release-worker.yml
+++ b/.github/workflows/release-worker.yml
@@ -74,6 +74,13 @@ jobs:
           -gs .github/settings.xml \
           -Pstagingrelease \
           -DgenerateBackupPoms=false \
+          -Dproperty=dataflow.version \
+          -DnewVersion=${BUILD_ZOO_HANDLER_spring_cloud_dataflow_version} \
+          -B             
+        jfrog rt mvn versions:set-property \
+          -gs .github/settings.xml \
+          -Pstagingrelease \
+          -DgenerateBackupPoms=false \
           -Dproperty=spring-cloud-deployer.version \
           -DnewVersion=${BUILD_ZOO_HANDLER_spring_cloud_deployer_version} \
           -B

--- a/spring-cloud-dataflow-build/spring-cloud-dataflow-build-dependencies/pom.xml
+++ b/spring-cloud-dataflow-build/spring-cloud-dataflow-build-dependencies/pom.xml
@@ -42,7 +42,6 @@
 	</properties>
 	<dependencyManagement>
 		<dependencies>
-
 			<dependency>
 				<groupId>net.minidev</groupId>
 				<artifactId>json-smart</artifactId>
@@ -81,7 +80,6 @@
 				<type>pom</type>
 				<scope>import</scope>
 			</dependency>
-
 			<dependency>
 				<groupId>io.fabric8</groupId>
 				<artifactId>kubernetes-client-bom</artifactId>

--- a/spring-cloud-dataflow-build/spring-cloud-dataflow-dependencies-parent/pom.xml
+++ b/spring-cloud-dataflow-build/spring-cloud-dataflow-dependencies-parent/pom.xml
@@ -36,6 +36,8 @@
 	</developers>
 	<properties>
 		<project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
+		<!-- NOTE: this needs to be updated and kept in sync w/ the project.version -->
+		<dataflow.version>2.11.1-SNAPSHOT</dataflow.version>
 	</properties>
 	<profiles>
 		<profile>

--- a/spring-cloud-dataflow-common/spring-cloud-dataflow-common-dependencies/pom.xml
+++ b/spring-cloud-dataflow-common/spring-cloud-dataflow-common-dependencies/pom.xml
@@ -21,17 +21,17 @@
 			<dependency>
 				<groupId>org.springframework.cloud</groupId>
 				<artifactId>spring-cloud-dataflow-common-flyway</artifactId>
-				<version>${project.version}</version>
+				<version>${dataflow.version}</version>
 			</dependency>
 			<dependency>
 				<groupId>org.springframework.cloud</groupId>
 				<artifactId>spring-cloud-dataflow-common-test-docker</artifactId>
-				<version>${project.version}</version>
+				<version>${dataflow.version}</version>
 			</dependency>
 			<dependency>
 				<groupId>org.springframework.cloud</groupId>
 				<artifactId>spring-cloud-dataflow-common-test-docker-junit5</artifactId>
-				<version>${project.version}</version>
+				<version>${dataflow.version}</version>
 			</dependency>
 		</dependencies>
 	</dependencyManagement>

--- a/spring-cloud-dataflow-dependencies/pom.xml
+++ b/spring-cloud-dataflow-dependencies/pom.xml
@@ -117,6 +117,21 @@
 				<artifactId>spring-cloud-dataflow-test</artifactId>
 				<version>${project.version}</version>
 			</dependency>
+			<dependency>
+				<groupId>org.springframework.cloud</groupId>
+				<artifactId>spring-cloud-common-security-config-core</artifactId>
+				<version>${project.version}</version>
+			</dependency>
+			<dependency>
+				<groupId>org.springframework.cloud</groupId>
+				<artifactId>spring-cloud-common-security-config-web</artifactId>
+				<version>${project.version}</version>
+			</dependency>
+			<dependency>
+				<groupId>org.springframework.cloud</groupId>
+				<artifactId>spring-cloud-starter-common-security-config-web</artifactId>
+				<version>${project.version}</version>
+			</dependency>
 		</dependencies>
 	</dependencyManagement>
 	<distributionManagement>

--- a/spring-cloud-dataflow-dependencies/pom.xml
+++ b/spring-cloud-dataflow-dependencies/pom.xml
@@ -19,118 +19,118 @@
 			<dependency>
 				<groupId>org.springframework.cloud</groupId>
 				<artifactId>spring-cloud-dataflow-shell</artifactId>
-				<version>${project.version}</version>
+				<version>${dataflow.version}</version>
 			</dependency>
 			<dependency>
 				<groupId>org.springframework.cloud</groupId>
 				<artifactId>spring-cloud-dataflow-shell-core</artifactId>
-				<version>${project.version}</version>
+				<version>${dataflow.version}</version>
 			</dependency>
 			<dependency>
 				<groupId>org.springframework.cloud</groupId>
 				<artifactId>spring-cloud-dataflow-completion</artifactId>
-				<version>${project.version}</version>
+				<version>${dataflow.version}</version>
 			</dependency>
 			<dependency>
 				<groupId>org.springframework.cloud</groupId>
 				<artifactId>spring-cloud-dataflow-core-dsl</artifactId>
-				<version>${project.version}</version>
+				<version>${dataflow.version}</version>
 			</dependency>
 			<dependency>
 				<groupId>org.springframework.cloud</groupId>
 				<artifactId>spring-cloud-dataflow-core</artifactId>
-				<version>${project.version}</version>
+				<version>${dataflow.version}</version>
 			</dependency>
 			<dependency>
 				<groupId>org.springframework.cloud</groupId>
 				<artifactId>spring-cloud-dataflow-rest-client</artifactId>
-				<version>${project.version}</version>
+				<version>${dataflow.version}</version>
 			</dependency>
 			<dependency>
 				<groupId>org.springframework.cloud</groupId>
 				<artifactId>spring-cloud-dataflow-configuration-metadata</artifactId>
-				<version>${project.version}</version>
+				<version>${dataflow.version}</version>
 			</dependency>
 			<dependency>
 				<groupId>org.springframework.cloud</groupId>
 				<artifactId>spring-cloud-dataflow-rest-resource</artifactId>
-				<version>${project.version}</version>
+				<version>${dataflow.version}</version>
 			</dependency>
 			<dependency>
 				<groupId>org.springframework.cloud</groupId>
 				<artifactId>spring-cloud-dataflow-registry</artifactId>
-				<version>${project.version}</version>
+				<version>${dataflow.version}</version>
 			</dependency>
 			<dependency>
 				<groupId>org.springframework.cloud</groupId>
 				<artifactId>spring-cloud-dataflow-container-registry</artifactId>
-				<version>${project.version}</version>
+				<version>${dataflow.version}</version>
 			</dependency>
 			<dependency>
 				<groupId>org.springframework.cloud</groupId>
 				<artifactId>spring-cloud-dataflow-composed-task-runner</artifactId>
-				<version>${project.version}</version>
+				<version>${dataflow.version}</version>
 			</dependency>
 			<dependency>
 				<groupId>org.springframework.cloud</groupId>
 				<artifactId>spring-cloud-dataflow-server-core</artifactId>
-				<version>${project.version}</version>
+				<version>${dataflow.version}</version>
 			</dependency>
 			<dependency>
 				<groupId>org.springframework.cloud</groupId>
 				<artifactId>spring-cloud-dataflow-server</artifactId>
-				<version>${project.version}</version>
+				<version>${dataflow.version}</version>
 			</dependency>
 			<dependency>
 				<groupId>org.springframework.cloud</groupId>
 				<artifactId>spring-cloud-dataflow-autoconfigure</artifactId>
-				<version>${project.version}</version>
+				<version>${dataflow.version}</version>
 			</dependency>
 			<dependency>
 				<groupId>org.springframework.cloud</groupId>
 				<artifactId>spring-cloud-starter-dataflow-server</artifactId>
-				<version>${project.version}</version>
+				<version>${dataflow.version}</version>
 			</dependency>
 			<dependency>
 				<groupId>org.springframework.cloud</groupId>
 				<artifactId>spring-cloud-starter-dataflow-server</artifactId>
-				<version>${project.version}</version>
+				<version>${dataflow.version}</version>
 				<type>test-jar</type>
 			</dependency>
 			<dependency>
 				<groupId>org.springframework.cloud</groupId>
 				<artifactId>spring-cloud-dataflow-audit</artifactId>
-				<version>${project.version}</version>
+				<version>${dataflow.version}</version>
 			</dependency>
 			<dependency>
 				<groupId>org.springframework.cloud</groupId>
 				<artifactId>spring-cloud-dataflow-platform-kubernetes</artifactId>
-				<version>${project.version}</version>
+				<version>${dataflow.version}</version>
 			</dependency>
 			<dependency>
 				<groupId>org.springframework.cloud</groupId>
 				<artifactId>spring-cloud-dataflow-platform-cloudfoundry</artifactId>
-				<version>${project.version}</version>
+				<version>${dataflow.version}</version>
 			</dependency>
 			<dependency>
 				<groupId>org.springframework.cloud</groupId>
 				<artifactId>spring-cloud-dataflow-test</artifactId>
-				<version>${project.version}</version>
+				<version>${dataflow.version}</version>
 			</dependency>
 			<dependency>
 				<groupId>org.springframework.cloud</groupId>
 				<artifactId>spring-cloud-common-security-config-core</artifactId>
-				<version>${project.version}</version>
+				<version>${dataflow.version}</version>
 			</dependency>
 			<dependency>
 				<groupId>org.springframework.cloud</groupId>
 				<artifactId>spring-cloud-common-security-config-web</artifactId>
-				<version>${project.version}</version>
+				<version>${dataflow.version}</version>
 			</dependency>
 			<dependency>
 				<groupId>org.springframework.cloud</groupId>
 				<artifactId>spring-cloud-starter-common-security-config-web</artifactId>
-				<version>${project.version}</version>
+				<version>${dataflow.version}</version>
 			</dependency>
 		</dependencies>
 	</dependencyManagement>
@@ -139,7 +139,7 @@
 		<site>
 			<id>spring-docs</id>
 			<url>
-				scp://static.springframework.org/var/www/domains/springframework.org/static/htdocs/spring-cloud/docs/${project.artifactId}/${project.version}
+				scp://static.springframework.org/var/www/domains/springframework.org/static/htdocs/spring-cloud/docs/${project.artifactId}/${dataflow.version}
 			</url>
 		</site>
 		<repository>

--- a/spring-cloud-dataflow-parent/pom.xml
+++ b/spring-cloud-dataflow-parent/pom.xml
@@ -134,6 +134,21 @@
 			</dependency>
 			<dependency>
 				<groupId>org.springframework.cloud</groupId>
+				<artifactId>spring-cloud-common-security-config-core</artifactId>
+				<version>${dataflow.version}</version>
+			</dependency>
+			<dependency>
+				<groupId>org.springframework.cloud</groupId>
+				<artifactId>spring-cloud-common-security-config-web</artifactId>
+				<version>${dataflow.version}</version>
+			</dependency>
+			<dependency>
+				<groupId>org.springframework.cloud</groupId>
+				<artifactId>spring-cloud-starter-common-security-config-web</artifactId>
+				<version>${dataflow.version}</version>
+			</dependency>
+			<dependency>
+				<groupId>org.springframework.cloud</groupId>
 				<artifactId>spring-cloud-task-dependencies</artifactId>
 				<version>${spring-cloud-task.version}</version>
 				<type>pom</type>

--- a/spring-cloud-dataflow-parent/pom.xml
+++ b/spring-cloud-dataflow-parent/pom.xml
@@ -11,15 +11,16 @@
 	<packaging>pom</packaging>
 	<url>https://cloud.spring.io/spring-cloud-dataflow/</url>
 	<properties>
+		<!-- NOTE: this needs to be updated and kept in sync w/ the project.version -->
+		<dataflow.version>2.11.1-SNAPSHOT</dataflow.version>
 		<java.version>1.8</java.version>
 		<javadoc.opts>-Xdoclint:none</javadoc.opts>
 		<maven-resources-plugin.version>3.3.1</maven-resources-plugin.version>
 		<spring-boot.version>2.7.16</spring-boot.version>
 		<spring.version>5.3.30</spring.version>
 		<spring-cloud-dataflow-ui.version>3.4.1-SNAPSHOT</spring-cloud-dataflow-ui.version>
-		<dataflow.version>${project.version}</dataflow.version>
-		<spring-cloud-dataflow-common.version>${project.version}</spring-cloud-dataflow-common.version>
-		<spring-cloud-skipper.version>${project.version}</spring-cloud-skipper.version>
+		<spring-cloud-dataflow-common.version>${dataflow.version}</spring-cloud-dataflow-common.version>
+		<spring-cloud-skipper.version>${dataflow.version}</spring-cloud-skipper.version>
 		<spring-cloud-deployer.version>2.9.1-SNAPSHOT</spring-cloud-deployer.version>
 		<spring-cloud-task.version>2.4.6</spring-cloud-task.version>
 		<spring-cloud-common-security-config.version>${dataflow.version}</spring-cloud-common-security-config.version>
@@ -211,16 +212,6 @@
 				<groupId>org.springframework.cloud</groupId>
 				<artifactId>spring-cloud-deployer-cloudfoundry</artifactId>
 				<version>${spring-cloud-deployer.version}</version>
-			</dependency>
-			<dependency>
-				<groupId>org.springframework.cloud</groupId>
-				<artifactId>spring-cloud-common-security-config-web</artifactId>
-				<version>${project.version}</version>
-			</dependency>
-			<dependency>
-				<groupId>org.springframework.cloud</groupId>
-				<artifactId>spring-cloud-starter-common-security-config-web</artifactId>
-				<version>${project.version}</version>
 			</dependency>
 			<dependency>
 				<groupId>org.springframework.cloud</groupId>

--- a/spring-cloud-skipper/spring-cloud-skipper-dependencies/pom.xml
+++ b/spring-cloud-skipper/spring-cloud-skipper-dependencies/pom.xml
@@ -19,42 +19,42 @@
 			<dependency>
 				<groupId>org.springframework.cloud</groupId>
 				<artifactId>spring-cloud-skipper</artifactId>
-				<version>${project.version}</version>
+				<version>${dataflow.version}</version>
 			</dependency>
 			<dependency>
 				<groupId>org.springframework.cloud</groupId>
 				<artifactId>spring-cloud-skipper-server-core</artifactId>
-				<version>${project.version}</version>
+				<version>${dataflow.version}</version>
 			</dependency>
 			<dependency>
 				<groupId>org.springframework.cloud</groupId>
 				<artifactId>spring-cloud-skipper-shell-commands</artifactId>
-				<version>${project.version}</version>
+				<version>${dataflow.version}</version>
 			</dependency>
 			<dependency>
 				<groupId>org.springframework.cloud</groupId>
 				<artifactId>spring-cloud-skipper-client</artifactId>
-				<version>${project.version}</version>
+				<version>${dataflow.version}</version>
 			</dependency>
 			<dependency>
 				<groupId>org.springframework.cloud</groupId>
 				<artifactId>spring-cloud-skipper-platform-kubernetes</artifactId>
-				<version>${project.version}</version>
+				<version>${dataflow.version}</version>
 			</dependency>
 			<dependency>
 				<groupId>org.springframework.cloud</groupId>
 				<artifactId>spring-cloud-skipper-platform-cloudfoundry</artifactId>
-				<version>${project.version}</version>
+				<version>${dataflow.version}</version>
 			</dependency>
 			<dependency>
 				<groupId>org.springframework.cloud</groupId>
 				<artifactId>spring-cloud-skipper-autoconfigure</artifactId>
-				<version>${project.version}</version>
+				<version>${dataflow.version}</version>
 			</dependency>
 			<dependency>
 				<groupId>org.springframework.cloud</groupId>
 				<artifactId>spring-cloud-starter-skipper-server</artifactId>
-				<version>${project.version}</version>
+				<version>${dataflow.version}</version>
 			</dependency>
 		</dependencies>
 	</dependencyManagement>

